contrib/drivers/huawei/dorado/metrics.go

```diff
@@ -0,0 +1,311 @@
+	//validate metric support list
+	supportedMetrics, err := d.GetMetricList(MetricResourceTypeController)
+	if supportedMetrics == nil {
+		log.Infof("no metrics found in the supported metric list")
+	}
+	controllers, err := d.client.ListControllers()
+	if err != nil {
+		log.Errorf("get controller failed: %s", err)
+		return nil, err
+	}
+
+	var tempMetricArray []*model.MetricSpec
+	for _, controller := range controllers {
+		// TODO: the controller id need to be optional
+		uuid := ObjectTypeController + ":" + controller.Id
+		metricMap, err := d.CollectPerformanceMetrics(uuid, supportedMetrics)
+		if err != nil {
+			log.Errorf("get performance data failed: %s", err)
+			return nil, err
+		}
+
+		for _, element := range supportedMetrics {
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     metricMap[element],
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+			metric := &model.MetricSpec{
+				InstanceID:   uuid,
+				InstanceName: uuid,
+				Job:          "HuaweiOceanStor",
+				Labels:       map[string]string{"device": MetricResourceTypeController},
+				Component:    MetricResourceTypeController,
+				Name:         element,
+				Unit:         metricToUnitMap[element],
+				AggrType:     "",
+				MetricValues: metricValues,
+			}
+			tempMetricArray = append(tempMetricArray, metric)
+		}
+	}
+	return tempMetricArray, nil
+}
+
+func (d *MetricDriver) CollectPoolMetrics() ([]*model.MetricSpec, error) {
+	// get Metrics to unit map
+	metricToUnitMap := getMetricToUnitMap(MetricResourceTypePool)
+	//validate metric support list
+	supportedMetrics, err := d.GetMetricList(MetricResourceTypePool)
+	if supportedMetrics == nil {
+		log.Infof("no metrics found in the supported metric list")
+	}
+
+	poolAll, err := d.client.ListStoragePools()
+	if err != nil {
+		log.Errorf("get controller failed: %s", err)
+		return nil, err
+	}
+	// Filter unsupported pools
+	var pools []StoragePool
+	for _, p := range poolAll {
+		if _, ok := d.conf.Pool[p.Name]; !ok {
+			continue
+		}
+		pools = append(pools, p)
+	}
+
+	var tempMetricArray []*model.MetricSpec
+	for _, pool := range pools {
+		// TODO: the controller id need to be optional
+		resId := ObjectTypePool + ":" + pool.Id
+		metricMap, err := d.CollectPerformanceMetrics(resId, supportedMetrics)
+		if err != nil {
+			log.Errorf("get performance data failed: %s", err)
+			return nil, err
+		}
+		poolId := uuid.NewV5(uuid.NamespaceOID, pool.Name).String()
+		for _, element := range supportedMetrics {
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     metricMap[element],
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+			metric := &model.MetricSpec{
+				InstanceID:   poolId,
+				InstanceName: pool.Name,
+				Job:          "HuaweiOceanStor",
+				Labels:       map[string]string{"device": MetricResourceTypePool},
```

Comment:
I suggest poolid as the lable here 
Labels:       map[string]string{"pool": poolId},